meshio/mdpa_io.py

```diff
@@ -0,0 +1,591 @@
+    "Line2D3": "line3",
+    "Triangle2D6": "triangle6",
+    "Triangle3D6": "triangle6",
+    "Quadrilateral2D9": "quad9",
+    "Quadrilateral3D9": "quad9",
+    "Tetrahedra3D10": "tetra10",
+    "Hexahedra3D27": "hexahedron27",
+    "Point2D": "vertex",
+    "Point3D": "vertex",
+    "Quadrilateral2D8": "quad8",
+    "Quadrilateral3D8": "quad8",
+    "Hexahedra3D20": "hexahedron20",
+}
+
+_meshio_to_mdpa_type = {
+    "line": "Line2D2",
+    "triangle": "Triangle2D3",
+    "quad": "Quadrilateral2D4",
+    "tetra": "Tetrahedra3D4",
+    "hexahedron": "Hexahedra3D8",
+    "wedge": "Prism3D6",
+    "line3": "Line2D3",
+    "triangle6": "Triangle2D6",
+    "quad9": "Quadrilateral2D9",
+    "tetra10": "Tetrahedra3D10",
+    "hexahedron27": "Hexahedra3D27",
+    "vertex": "Point2D",
+    "quad8": "Quadrilateral2D8",
+    "hexahedron20": "Hexahedra3D20",
+}
+inverse_num_nodes_per_cell = {v: k for k, v in num_nodes_per_cell.items()}
+
+local_dimension_types = {
+    "Line2D2": 1,
+    "Line3D2": 1,
+    "Triangle2D3": 2,
+    "Triangle3D3": 2,
+    "Quadrilateral2D4": 2,
+    "Quadrilateral3D4": 2,
+    "Tetrahedra3D4": 3,
+    "Hexahedra3D8": 3,
+    "Prism3D6": 3,
+    "Line2D3": 1,
+    "Triangle2D6": 2,
+    "Triangle3D6": 2,
+    "Quadrilateral2D9": 2,
+    "Quadrilateral3D9": 2,
+    "Tetrahedra3D10": 3,
+    "Hexahedra3D27": 3,
+    "Point2D": 0,
+    "Point3D": 0,
+    "Quadrilateral2D8": 2,
+    "Quadrilateral3D8": 2,
+    "Hexahedra3D20": 3,
+}
+
+
+def read(filename):
+    """Reads a KratosMultiphysics mdpa file.
+    """
+    # if (have_kratos is True): # TODO: Implement natively
+    # pass
+    # else:
+    with open(filename, "rb") as f:
+        mesh = read_buffer(f)
+    return mesh
+
+
+def _read_nodes(f, is_ascii, data_size):
+    # The first line is the number of nodes
+    pos = f.tell()
+    lines = f.readlines()
+    num_nodes = 0
+    for line in lines:
+        str_line = str(line)
+        if "End Nodes" in str_line:
+            break
+        num_nodes += 1
+    f.seek(pos)
+
+    points = numpy.fromfile(f, count=num_nodes * 4, sep=" ").reshape((num_nodes, 4))
+    # The first number is the index
+    points = points[:, 1:]
+
+    line = f.readline().decode("utf-8")
+    assert line.strip() == "End Nodes"
+    return points
+
+
+def _read_cells(f, cells, is_ascii, cell_tags, environ=None):
+    # First we try to identity the entity
+    t = None
+    if environ is not None:
+        if "Begin Elements" in environ:
+            entity_name = environ.replace("Begin Elements", "")
+            for key in _mdpa_to_meshio_type:
+                if key in entity_name:
+                    t = _mdpa_to_meshio_type[key]
+                    break
+        elif "Begin Conditions" in environ:
+            entity_name = environ.replace("Begin Conditions", "")
+            for key in _mdpa_to_meshio_type:
+                if key in entity_name:
+                    t = _mdpa_to_meshio_type[key]
+                    break
```

Comment:
`else`? `raise` or `pass`?

Reply:
There is an automatic check later (in case of not assigned and the type is None, then it takes the type from the list). The problem with that is that maybe it identifies a triangle as a a quadratic line. That's why I do the check